widget/struct_form.go

```diff
@@ -0,0 +1,146 @@
+package widget
+
+import (
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/data/binding"
+	"fyne.io/fyne/v2/widget"
+)
+
+type StructForm struct {
+	widget.BaseWidget
+	structType reflect.Type
+	modal      *widget.PopUp
+	canvas     fyne.Canvas
```

Comment:
You do not need to store this value if you only access it from within NewStructForm.